pmd-java/src/main/resources/category/java/design.xml

```diff
@@ -50,21 +50,50 @@ public abstract class Example {
     <rule name="AvoidCatchingGenericException"
           since="4.2.6"
           language="java"
-          message="Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block"
+          message="Avoid catching {0} in try-catch block"
           class="net.sourceforge.pmd.lang.rule.xpath.XPathRule"
           externalInfoUrl="${pmd.website.baseurl}/pmd_rules_java_design.html#avoidcatchinggenericexception">
         <description>
-Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block.
+Avoid catching generic exceptions in try-catch blocks. Catching overly broad exception types makes it difficult to understand what can actually go wrong in your code and can hide real problems.
+
+**Why these exceptions should not be caught:**
+
+* **Exception**: This is the base class for all checked exceptions. Catching it means you're handling all possible checked exceptions the same way, which is rarely appropriate and makes error handling less precise.
```

Comment:
```suggestion
* **Exception**: This is the base class for all checked exceptions and for all unchecked exceptions. Catching it means you're handling all possible checked/unchecked exceptions the same way, which is rarely appropriate and makes error handling less precise. As this implicitly catches "RuntimeException" as well, it can hide bugs.
```